### PR TITLE
add fix for ValueError when there are multiple classes or outputs

### PR DESCRIPTION
**Pull request by quantum-mango**
### Summary
This PR fixes a ValueError in the MultiClassObjectiveFunction that occurs when a neural network needs to predict multiple classes or multiple outputs. The loss function was being passed the entire y values instead of the current iteration's target column within the y values.

### Details and comments

See also https://qiskit.slack.com/archives/CB6C24TPB/p1686792241848319?thread_ts=1686282201.761999&cid=CB6C24TPB

### CODE REVIEW

**Comment by adekusar-drl** [comment]
Closing as there's no progress.
